RIOT-OS/RIOT
# WIP: tests/ieee802154_hal: extend radio hal test

### Contribution description
This PR adds multiple functions for the IEEE 802.15.4

- spam:  Sends as fast and as many packets as specified in the parameters
- reply: The receiver sends the package back as soon as possible
- enable_prints: Prints interesting stuff. e.g. received packet. May has to be disabled due to timer issues caused by the printing
- test_channel: Sends and receive on different channels
- check_last_packet: Checks if the last packet matches to the expect sender




### Testing procedure
There will be a automated test in riotctrl.

For manual testing you need two boards and flash it with tests/ieee802154_hal

Usage of the function spam:
Call spam on board with the parameters
spam <address_board2> <length_of_the_payload> <number_of_packets> <delay>

If you call spam with the parameters spam <address_board2> 5 3 1
The device will send a packet to device2 every 1 ms with a payload of 3 Chars in total 5 packets will be send. 

You can set device2 in reply mode, means it will sends every packet back he receives, by calling the function reply. Just make sure to be easy on the delay otherwise the radio won't be able to handle this. You may disable the printing by calling the function enable_prints.




### Issues/PRs references
None
